components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/edit.jsp

```diff
@@ -231,61 +244,51 @@
+                    var oidcLogoutUrl = document.getElementById('logout_url').value;
 
                     if (callbackUrl.indexOf("#") !== -1) {
                         CARBON.showWarningDialog('<fmt:message key="callback.is.fragment"/>');
                         return false;
                     }
-                    if (backChannelLogoutUrl.indexOf("#") !== -1) {
-                        CARBON.showWarningDialog('<fmt:message key="backchannel.logout.is.fragment"/>');
+                    if (oidcLogoutUrl.indexOf("#") !== -1) {
+                        CARBON.showWarningDialog('<fmt:message key="logout.url.is.fragment"/>');
                         return false;
                     }
                     var value = document.getElementsByName("application")[0].value;
                     if (value == '') {
                         CARBON.showWarningDialog('<fmt:message key="application.is.required"/>');
                         return false;
                     }
+
                     var versionValue = document.getElementsByName("oauthVersion")[0].value;
-                    if (versionValue == '<%=OAuthConstants.OAuthVersions.VERSION_2%>') {
+
+                    if (versionValue === '<%= OAuthConstants.OAuthVersions.VERSION_2%>') {
                         if (!$(jQuery("#grant_authorization_code"))[0].checked && !$(jQuery("#grant_implicit"))[0].checked) {
                             document.getElementsByName("callback")[0].value = '';
                         } else {
                             // This is to support providing regex patterns for callback URLs
                             if (callbackUrl.startsWith("regexp=")) {
                                 // skip validation
-                            } else if (!isWhiteListed(callbackUrl, ["url"])) {
+                            } else if (!(isWhiteListed(callbackUrl, ["https-url"]) || isWhiteListed(callbackUrl, ["http-url"])) || !isNotBlackListed(callbackUrl, ["uri-unsafe-exists"])) {
                                 CARBON.showWarningDialog('<fmt:message key="callback.is.not.url"/>');
                                 return false;
                             }
-                            if (!isWhiteListed(backChannelLogoutUrl, ["url"]) || !isNotBlackListed(backChannelLogoutUrl,
+
+                            if (oidcLogoutType === "<%= OAuthConstants.OIDCConfigProperties.BACK_CHANNEL_LOGOUT_SELECTED%>") {
+                                if (!((isWhiteListed(oidcLogoutUrl, ["https-url"]) || isWhiteListed(oidcLogoutUrl, ["http-url"]))) || !isNotBlackListed(oidcLogoutUrl,
                                     ["uri-unsafe-exists"])) {
-                                CARBON.showWarningDialog('<fmt:message key="backchannel.logout.is.not.url"/>');
-                                return false;
-                            }
-                            if (!isWhiteListed(userTokenExpiryTime, ["digits-only"])) {
```

Comment:
Any reasons for removing these ?

Reply:
The blocks were moved outside the if block as those checks need to done for every OAuth 2 application scenario.